nvflare/private/fed/client/client_engine.py

```diff
@@ -219,7 +220,9 @@ def shutdown(self) -> str:
         touch_file = os.path.join(self.args.workspace, "shutdown.fl")
         self.fire_event(EventType.SYSTEM_END, self.new_context())
 
-        _ = self.executor.submit(lambda p: _shutdown_client(*p), [self.client, self.admin_agent, touch_file])
+        # _ = self.executor.submit(lambda p: _shutdown_client(*p), [self.client, self.admin_agent, touch_file])
+        thread = threading.Thread(target=_shutdown_client, args=(self.client, self.admin_agent, touch_file))
+        thread.start()
```

Comment:
why do we make this change?

what about `do_validate` function?

Should we change all places that use `self.executor` inside this client_engine.py?

Reply:
no need to use the ThreadPoolExecutor here. Also removed the no longer needed do_validate().